### PR TITLE
Improve search performance to validGeneList

### PR DESCRIPTION
**Pull request by amaraim22**

### CODE REVIEW

File: main/html/index.html [hunk 1]
```diff
@@ -175,23 +178,41 @@
...
-                  // If the gene entered is not valid, don't create new tag:
-                  if (
-                    ~validGeneList.includes(params.term.toUpperCase()) === -1
-                  ) {
-                    return null;
+                // creating query with pagination functionality.
+                query: function (data) {
+                  var pageSize,
+                          dataset,
+                          that = this;
```
Comment (ZhenghaoZhu) [inline main/html/index.html:201]: "that" is a bit abstract of a variable name, can you change it to a more descriptive name?

File: main/html/index.html [hunk 2]
```diff
@@ -175,23 +177,41 @@
...
+                  pageSize = 5; // Number of the option loads at a time
+                  results = [];
+                  if (data.term && data.term !== '') {
+                    // HEADS UP; for the _.filter function I use underscore (actually lo-dash) here
+                    results = _.filter(that.data, function (e) {
+                      return e.text.toUpperCase().indexOf(data.term.toUpperCase()) >= 0;
+                    });
+                  } else if (data.term === '') {
```
Comment (ericr491) [inline main/html/index.html:209]: Is 'else if' necessary here? Looks like 'else' would work here.

File: main/html/index.html [hunk 3]
```diff
@@ -202,10 +222,31 @@
...
```
Comment (ericr491) [inline main/html/index.html:241]: See above

**Review by enemeth19 — COMMENTED** [review]
This looks really great, @amaraim22 ! Just consider the changes suggested by @ZhenghaoZhu and @ericr491, make those changes, push to the same branch, and then we will merge this branch into dev :)

**Comment by amaraim22** [comment]
I think this should be good. please let me know if you guys have any more suggestions/changes

**Review by ZhenghaoZhu — APPROVED** [review]

**Pull request merged**